### PR TITLE
LF: minor cleanup in SortedLookupList

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Hasher.scala [hunk 1]
```diff
@@ -1,101 +0,0 @@
...
-package com.daml.lf.speedy.svalue
-
-import com.daml.lf.speedy.SValue
-import com.daml.lf.speedy.SValue._
-
-import scala.annotation.tailrec
-import scala.collection.JavaConverters._
-import scala.util.hashing.MurmurHash3
-
-private[speedy] object Hasher {
```
Comment (SamirTalwar-DA) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Hasher.scala:13]: I assume this is unrelated to the other stuff?

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala [hunk 2]
```diff
@@ -48,34 +49,34 @@ final class SortedLookupList[+X] private (entries: ImmArray[(String, X)]) extend
...
+    }
+
+  def fromOrderedImmArray[X](entries: ImmArray[(String, X)]): Either[String, SortedLookupList[X]] =
+    nonOrderedEntry(entries) match {
+      case None => Right(new SortedLookupList(entries))
+      case Some(_) => Left(s"the entries $entries is not sorted by key")
+    }
 
   def fromOrderedIterator[X](entries: Iterator[(String, X)]): Either[String, SortedLookupList[X]] =
     fromOrderedImmArray(entries.to[ImmArray])
```
Comment (SamirTalwar-DA) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala:74]: I still think the other functions could work on `Iterator[(String, X)]`. If you go back to using `sliding` you only need one.

**Review by SamirTalwar-DA — APPROVED** [review]
Looks good, though I think more iterators, fewer arrays. 😉

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Hasher.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Hasher.scala]: Yes. fixed.

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala:74]: I do not get it. Please feel free to improve my code ;).

Comment (SamirTalwar-DA) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala:74]: Hmmm… looks like you can probably just delete this function entirely. It's unused. Does it exist for a reason?

Comment (remyhaemmerle-da) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala:74]: Good catch. I was using it in the first version of #7334, but apparently I dropped it.

Comment (remyhaemmerle-da) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala:74]: pruned

Comment (SamirTalwar-DA) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala:74]: And pushed? 🙃

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala [hunk 3]
```diff
@@ -48,34 +49,31 @@ final class SortedLookupList[+X] private (entries: ImmArray[(String, X)]) extend
...
+  def fromImmArray[X](entries: ImmArray[(String, X)]): Either[String, SortedLookupList[X]] =
+    nonOrderedEntry(entries.toSeq.sorted(EntryOrdering).toImmArray) match {
+      case None => Right(new SortedLookupList(entries))
+      case Some((key, _)) => Left(s"key $key duplicated when trying to build map")
+    }
+
+  def fromOrderedImmArray[X](entries: ImmArray[(String, X)]): Either[String, SortedLookupList[X]] =
+    nonOrderedEntry(entries) match {
+      case None => Right(new SortedLookupList(entries))
+      case Some(_) => Left(s"the entries $entries is not sorted by key")
```
Comment (hurryabit) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala:70]:
```suggestion
      case Some(_) => Left(s"the entries $entries are not sorted by key")
```

**Review by hurryabit — APPROVED** [review]
This looks reasonable to me. Thanks.
